### ISSUE TITLE
#![allow(clippy::too_long_first_doc_paragraph)] in lib.rs is ignored

### ISSUE THREAD

**Issue by IceTDrinker** [rust-lang/rust-clippy#13758]
### Description

Some of our doc comments have long first paragraphs, which we explicitely opt-in to allow but we still get the clippy warnings

project where this is happening:

https://github.com/zama-ai/tfhe-rs/

```bash
RUSTFLAGS="-C target-cpu=native" cargo "+nightly-2024-11-29" clippy --all-targets \
        --features=x86_64-unix,boolean,shortint,integer,internal-keycache,zk-pok,strings \
        -p tfhe -- --no-deps -D warnings
```

### Version

```text
rustc 1.85.0-nightly (a2545fd6f 2024-11-28)
binary: rustc
commit-hash: a2545fd6fc66b4323f555223a860c451885d1d2b
commit-date: 2024-11-28
host: x86_64-unknown-linux-gnu
release: 1.85.0-nightly
LLVM version: 19.1.4
```

**Comment by IceTDrinker** [comment]
looks like the culprit could be

```
rustc 1.84.0-nightly (439284741 2024-10-21)
binary: rustc
commit-hash: 4392847410ddd67f6734dd9845f9742ff9e85c83
commit-date: 2024-10-21
host: x86_64-unknown-linux-gnu
release: 1.84.0-nightly
LLVM version: 19.1.1
```

**Comment by IceTDrinker** [comment]
Ah I see  we should warn then allow, last configuration being the one applied potentially

**Comment by IceTDrinker** [comment]
Re ordering seems to do the trick, it is logical when you think about it, but then maybe a warning/lint to warn first allow after or something about order of settings could be welcome. I'll close this

Thanks
